test/controllers/users_controller_test.rb

```diff
@@ -0,0 +1,36 @@
+require 'test_helper'
+
+class UsersControllerTest < ActionDispatch::IntegrationTest
+  test 'should get /users endpoint' do
+    get '/users'
+    assert_response :success
+  end
+
+  test '/users should retrieve all users as json' do
+    users = ['John', 'Polly']
+    users.each { |user| User.create(name: user)}
+
+    get '/users'
+
+    users_response = JSON.parse(@response.body, symbolize_names: true)
+    response_names = users_response.map {|user| user[:name]}
+
+    assert_equal response_names, users
+    assert_equal 2, users_response.count
+  end
+
+  test 'should get /users/1 endpoint' do
```

Comment:
I would probably put the tests for a single user ahead of the test for two users, however that's mostly because that's the order I would have implemented them.

It's the "Zom" part of Zombie testing: http://blog.wingman-sw.com/tdd-guided-by-zombies